k9mail/src/main/java/com/fsck/k9/activity/Accounts.java

```diff
@@ -1883,16 +1887,47 @@ public void onClick(View v) {
 
     }
 
-    public void onExport(final boolean includeGlobals, final Account account) {
+    public static final String EXTRA_INC_GLOBALS = "include_globals";
+    public static final String EXTRA_ACCOUNTS = "accountUuids";
 
+    public void onExport(final boolean includeGlobals, final Account account) {
         // TODO, prompt to allow a user to choose which accounts to export
-        Set<String> accountUuids = null;
+        ArrayList<String> accountUuids = null;
         if (account != null) {
-            accountUuids = new HashSet<String>();
+            accountUuids = new ArrayList<>();
             accountUuids.add(account.getUuid());
         }
 
-        ExportAsyncTask asyncTask = new ExportAsyncTask(this, includeGlobals, accountUuids);
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.KITKAT) {
+            Intent intent = new Intent(Intent.ACTION_CREATE_DOCUMENT);
+
+            intent.addCategory(Intent.CATEGORY_OPENABLE);
+            intent.setType("text/plain");
+            intent.putExtra(Intent.EXTRA_TITLE, SettingsExporter.EXPORT_FILENAME);
+            intent.putStringArrayListExtra(EXTRA_ACCOUNTS, accountUuids);
```

Comment:
The result Intent won't contain the extras specified in the request Intent. So this doesn't work.